File: src/main/java/com/learnspark/plan/entity/Task.java
package com.learnspark.plan.entity;

import com.learnspark.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * 学习任务实体，对应 tasks 表（1 阶段 : N 任务）。
 *
 * <p>支持软删除（deleted_at），状态流转：pending → submitted → passed/failed。
 */
@Entity
@Table(name = "tasks")
@SQLRestriction("deleted_at IS NULL")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task extends BaseEntity {

    @Id
    @Column(name = "id", updatable = false, columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "phase_id", columnDefinition = "VARCHAR(36)")
    private String phaseId;

    @Column(name = "project_id", nullable = false, columnDefinition = "VARCHAR(36)")
    private String projectId;

    @Column(name = "day_number")
    private Integer dayNumber;

    @Column(name = "title", length = 500)
    private String title;

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "verification_criteria", columnDefinition = "TEXT")
    private String verificationCriteria;

    @Column(name = "status", length = 20)
    private String status;

    @Column(name = "due_date")
    private LocalDate dueDate;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @PrePersist
    void fillDefaults() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.status == null) {
            this.status = "pending";
        }
        if (this.sortOrder == null) {
            this.sortOrder = 0;
        }
    }
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/entity/Task.kt
package com.learnspark.server.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.time.LocalDate

/**
 * 任务实体（按文档 §11.2.3：tasks 表）。
 *
 * 阶段 3.1：迁移目标。
 */
@Entity
@Table(name = "tasks")
class Task(
    @Id
    var id: String = "",

    @Column(name = "phase_id", nullable = false)
    var phaseId: String = "",

    @Column(nullable = false)
    var title: String = "",

    @Column(columnDefinition = "TEXT")
    var description: String? = null,

    @Column(name = "sort_order", nullable = false)
    var sortOrder: Int = 0,

    @Column(name = "estimated_hours", nullable = false)
    var estimatedHours: Int = 1,

    @Column(name = "actual_hours", nullable = false)
    var actualHours: Int = 0,

    @Column(nullable = false, length = 20)
    var status: String = "pending",

    @Column(name = "due_date")
    var dueDate: LocalDate? = null,

    @Column(nullable = false)
    var version: Long = 0,

    @Column(name = "created_at", insertable = false, updatable = false)
    var createdAt: Instant? = null,

    @Column(name = "updated_at", insertable = false, updatable = false)
    var updatedAt: Instant? = null,
)
